implementations/rust/ockam/ockam_app_lib/src/state/mod.rs

```diff
@@ -60,6 +64,10 @@ pub struct AppState {
     application_state_callback: ApplicationStateCallback,
     notification_callback: NotificationCallback,
     node_manager: Arc<RwLock<Arc<InMemoryNode>>>,
+    refresh_project_scheduler: Arc<Mutex<Option<Scheduler>>>,
+    refresh_invitations_scheduler: Arc<Mutex<Option<Scheduler>>>,
+    refresh_inlets_scheduler: Arc<Mutex<Option<Scheduler>>>,
```

Comment:
Have you considered using [cells](https://docs.rs/once_cell/latest/once_cell/#safe-initialization-of-global-data) here? Perhaps you could replace `Mutex<Option>` by `OnceCell<Scheduler>`.

The instantiation should be `refresh_project_scheduler: Arc::new(OnceCell<Scheduler>),`.


The initialization would look like `self.refresh_project_scheduler.set(Arc::new(Scheduler::create(...)))`.


And finally, using it would be as simple as `self.refresh_project_scheduler.get()` if you add a method like the `global` method described in the docs.

I haven't checked this in detail, but in case in can be helpful.

Reply:
this seems MUCH better!

Reply:
I ended up using `OnceLock` 👍

Reply:
Nice, never had the chance to see `OnceLock` in action yet, looks great!